aws/resource_aws_security_group_rule.go

```diff
@@ -62,6 +62,13 @@ func resourceAwsSecurityGroupRule() *schema.Resource {
 				Type:     schema.TypeInt,
 				Required: true,
 				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					protocol := protocolForValue(d.Get("protocol").(string))
```

Comment:
in the future it'd be good for `protocolForValue` to return a `*string` so we can easily detect this is invalid?